packages/aws-cdk-lib/aws-cloudfront/lib/private/cache-behavior.ts

```diff
@@ -22,10 +22,21 @@ export interface CacheBehaviorProps extends AddBehaviorOptions {
  * CloudFrontWebDistribution implementation.
  */
 export class CacheBehavior {
+  public readonly grpcEnabled?: boolean;
   private readonly originId: string;
 
   constructor(originId: string, private readonly props: CacheBehaviorProps) {
     this.originId = originId;
+    this.grpcEnabled = props.enableGrpc;
+
+    if (this.grpcEnabled) {
+      if (props.allowedMethods !== AllowedMethods.ALLOW_ALL) {
+        throw new Error('\'allowedMethods\' can only be AllowedMethods.ALLOW_ALL if \'enableGrpc\' is true');
+      }
+      if (props.edgeLambdas !== undefined && props.edgeLambdas.length > 0) {
+        throw new Error('\'edgeLambdas\' cannot be specified if \'enableGrpc\' is true');
+      }
```

Comment:
Otherwise CFn will fail

```
Resource handler returned message: "Invalid request provided: AWS::CloudFront::Distribution: The parameter gRPC doesn't support Lambda@Edge. (Service: CloudFront, Status Code: 400, Request ID:...
```